analyzere/base_resources.py

```diff
@@ -229,10 +229,17 @@ def upload_data(self, file_or_str):
         """
         file_obj = StringIO(file_or_str) if isinstance(
             file_or_str, six.string_types) else file_or_str
-        length = utils.file_length(file_obj)
 
-        # Initiate upload session
-        request_raw('post', self._data_path, headers={'Entity-Length': length})
+        # Upload file with known entity size if file object supports random
+        # access.
+        if hasattr(file_obj, 'seek'):
```

Comment:
It looks like [2.7 implements `seekable` ](https://docs.python.org/2/library/io.html#io.IOBase.seekable); am I missing something?